src/special/webpack.js

```diff
@@ -39,14 +42,82 @@ function getLoaders(deps, loaders) {
     .value();
 }
 
+function parseWebpack1(filepath, deps) {
+  const module = require(filepath).module || {}; // eslint-disable-line global-require
+  const loaders = getLoaders(deps, module.loaders);
+  const preLoaders = getLoaders(deps, module.preLoaders);
+  const postLoaders = getLoaders(deps, module.postLoaders);
+  return [...loaders, ...preLoaders, ...postLoaders];
+}
+
+function mapRuleLoaders(module) {
+  return module.rules
+    .filter(rule => rule.loaders)
+    .map(rule => rule.loaders
+      .map(loader => ({
+        loader,
+      })));
+}
+
+function mapRuleUse(module) {
+  return module.rules
+    // filter use or loader (loader is a shortcut to use)
+    .filter(rule => rule.use || rule.loader)
+    .map((rule) => {
+      const key = rule.use ? 'use' : 'loader';
+
+      // map simple strings as { loader: string }
```

Comment:
Instead of having three `if` clauses, I could suggest to convert the first two cases to `Array`, then handle it with the third case logic.